### ISSUE TITLE
js-function-in-arglist.d fails when configured with --disable-external-sort

### ISSUE THREAD

**Issue by k-takata** [universal-ctags/ctags#1411]
How to reproduce:

```sh
$ ./configure --disable-external-sort
$ make units CATEGORIES=parser-javascript.r UNITS=js-function-in-arglist
```

(I couldn't find description about the `CATEGORIES` setting in `docs/units.rst`. It should be documented.)

Output Diff:

```diff
@@ -1 +0,0 @@
-func1	input.js	/^    Test.func1 = function func1(){ console.log('func1'); };$/;"	f
@@ -3 +2 @@
-func2	input.js	/^    Test.func2 = function func2(){ console.log('func2'); };$/;"	f
+func1	input.js	/^    Test.func1 = function func1(){ console.log('func1'); };$/;"	f
@@ -4,0 +4 @@
+func2	input.js	/^    Test.func2 = function func2(){ console.log('func2'); };$/;"	f
```

The output is hard to read, so I paste the expected/actual output.
Expected:

```
func1	input.js	/^    Test.func1 = function func1(){ console.log('func1'); };$/;"	f
func1	input.js	/^    Test.func1 = function func1(){ console.log('func1'); };$/;"	f	function:Test
func2	input.js	/^    Test.func2 = function func2(){ console.log('func2'); };$/;"	f
func2	input.js	/^    Test.func2 = function func2(){ console.log('func2'); };$/;"	f	function:Test
funcGen	input.js	/^  function funcGen(A){$/;"	f
```

Actual:

```
func1	input.js	/^    Test.func1 = function func1(){ console.log('func1'); };$/;"	f	function:Test
func1	input.js	/^    Test.func1 = function func1(){ console.log('func1'); };$/;"	f
func2	input.js	/^    Test.func2 = function func2(){ console.log('func2'); };$/;"	f	function:Test
func2	input.js	/^    Test.func2 = function func2(){ console.log('func2'); };$/;"	f
funcGen	input.js	/^  function funcGen(A){$/;"	f
```

You can see that the line 1 and 2, and also the line 3 and 4 are swapped.

Reproduced on MSYS2 and Cygwin.  Also reproduced when I use MinGW and `mk_ming.mak`.
When I use VC and `mk_mvc.mak` locally, it also fails. I'm not sure why this doesn't fail on AppVeyor.
Not tested on other environments.

#1410 fails because of this.

**Comment by k-takata** [comment]
It seems that the function `internalSortTags()` sorts lines including the end of lines.

E.g. it compares the following two strings when sorting:

```
"func1\tinput.js\t/^    Test.func1 = function func1(){ console.log('func1'); };$/;\"\tf\n"
"func1\tinput.js\t/^    Test.func1 = function func1(){ console.log('func1'); };$/;\"\tf\tfunction:Test\n"
```

`\n` is grater than `\t`, therefore the line 1 and 2 are swapped.

**Comment by masatake** [comment]
How about replacing the last '\n' characters with '\0' in internalSortTags?
Here is my sketch.

```
diff --git a/main/sort.c b/main/sort.c
index 7b8d6cf..d5da9d8 100644
--- a/main/sort.c
+++ b/main/sort.c
@@ -193,7 +193,7 @@ static int compareTags (const void *const one, const void *const two)
 }
 
 static void writeSortedTags (
-		char **const table, const size_t numTags, const bool toStdout)
+		char **const table, const size_t numTags, const bool toStdout, bool newlineReplaced)
 {
 	MIO *mio;
 	size_t i;
@@ -214,8 +214,11 @@ static void writeSortedTags (
 		 *  pattern) if this is not an xref file.
 		 */
 		if (i == 0  ||  Option.xref  ||  strcmp (table [i], table [i-1]) != 0)
+		{
 			if (mio_puts (mio, table [i]) == EOF)
 				failedSort (mio, NULL);
+			else if (newlineReplaced)
+				mio_putc (mio, '\n');
 	}
 	if (toStdout)
 		mio_flush (mio);
@@ -228,6 +231,7 @@ extern void internalSortTags (const bool toStdout, MIO* mio, size_t numTags)
 	const char *line;
 	size_t i;
 	int (*cmpFunc)(const void *, const void *);
+	bool newlineReplaced = false;
 
 	/*  Allocate a table of line pointers to be sorted.
 	 */
@@ -260,6 +264,11 @@ extern void internalSortTags (const bool toStdout, MIO* mio, size_t numTags)
 				failedSort (mio, "out of memory");
 			DebugStatement ( mallocSize += stringSize; )
 			strcpy (table [i], line);
+			if (table[i][line] == '\n')
+			{
+				table[i][line] = '\0';
+				newlineReplaced = true;
+			}
 			++i;
 		}
 	}
@@ -270,7 +279,7 @@ extern void internalSortTags (const bool toStdout, MIO* mio, size_t numTags)
 	 */
 	qsort (table, numTags, sizeof (*table), cmpFunc);
 
-	writeSortedTags (table, numTags, toStdout);
+	writeSortedTags (table, numTags, toStdout, newlineReplaced);
 
 	PrintStatus (("sort memory: %ld bytes\n", (long) mallocSize));
 	for (i = 0 ; i < numTags ; ++i)
```

**Comment by k-takata** [comment]
I wrote a fix based on your comment. #1418

> When I use VC and mk_mvc.mak locally, it also fails. I'm not sure why this doesn't fail on AppVeyor.

I found that the test on AppVeyor was not worked as I expected. OMG!
I expected that a binary built by VC is tested by MSYS2, but actually `make check` overwrote the binary and a binary built by MSYS2 was tested. LOL
I will fix this in #1410.
